src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/Mutect2Engine.java

```diff
@@ -378,26 +369,15 @@ private boolean isReadFromNormal(final GATKRead rec) {
     }
 
     //TODO: move to SomaticGenotypingEngine with all the other annotations
-    final List<VariantContext> annotateCalls(final HaplotypeCallerGenotypingEngine.CalledHaplotypes calledHaplotypes) {
+    // TODO: can we do this with the Annotation classes instead?
+    final List<VariantContext> addEventCountAnnotation(final HaplotypeCallerGenotypingEngine.CalledHaplotypes calledHaplotypes) {
         final int eventCount = calledHaplotypes.getCalls().size();
 
-        final List<VariantContext> annotatedCalls = new ArrayList<>();
-        // can we do this with the Annotation classes instead?
-        for (final VariantContext originalVC : calledHaplotypes.getCalls()) {
-            final VariantContextBuilder vcb = new VariantContextBuilder(originalVC);
-
-            final Map<String, Object> attributes = new HashMap<>(originalVC.getAttributes());
+        return calledHaplotypes.getCalls().stream().map(vc -> {
```

Comment:
At any rate, I think this block of code belongs at the end of `SomaticGenotypingEngine.callMutation` (right before/after `final List<VariantContext> outputCalls = doPhysicalPhasing...`). It cannot be a standalone annotation class because there it wouldn't have access to the size of `calledHaplotypes`...I think.